webserver/views/api/v1/core.py

```diff
@@ -267,6 +267,93 @@ def get_many_highlevel():
     return jsonify(recording_details)
 
 
+def parse_select_features():
+    """Check whether the features are found or not.
+    Parse the query string of features to create a list of 
+    the json paths to features, excluding those that are 
+    not offered.
+
+    If features are missing, an APIBadRequest Exception is
+    raised stating the missing features message.
+    
+    Returns a string of feature paths with alias:
+        "llj.data->'feature1' AS feature1, ..., llj.data->'featureN' AS featureN"
+    """
+    features_param = request.args.get("features")
+    if not features_param:
+        raise webserver.views.api.exceptions.APIBadRequest("Missing `features` parameter")
+
+    selectable_features = ['lowlevel.average_loudness', 
+                            'lowlevel.dynamic_complexity', 
+                            'metadata.audio_properties.replay_gain', 
+                            'rhythm.beats_count', 
+                            'rhythm.beats_loudness.mean', 
+                            'rhythm.bpm', 
+                            'rhythm.bpm_histogram_first_peak_bpm.mean', 
+                            'rhythm.bpm_histogram_second_peak_bpm.mean', 
+                            'rhythm.danceability', 
+                            'rhythm.onset_rate', 
+                            'tonal.chords_key', 
+                            'tonal.chords_scale', 
+                            'tonal.key_key', 
+                            'tonal.key_scale', 
+                            'tonal.tuning_frequency', 
+                            'tonal.tuning_equal_tempered_deviation']
+
+    parsed_features = []
+    raw_paths = []
+    for feature in features_param.split(';'):
+        if feature in selectable_features:
+            raw_paths.append(feature)
+            # Build feature path
+            feature_path = 'llj.data'
+            for element in feature.split('.'):
+                feature_path += '->\'' + element + '\''
+            parsed_features.append(feature_path)
+
+    # Remove duplicates, preserving order
+    seen = set()
+    parsed_features = [x for x in parsed_features if not (x in seen or seen.add(x))]
+
+    features_string = parsed_features[0] + ' AS "' + raw_paths[0] + '", '
+    for feature, alias in zip(parsed_features[1:len(parsed_features)-1], raw_paths[1:len(raw_paths)-1]):
+        features_string += feature + ' AS "' +  alias + '", '
+    features_string += parsed_features[len(parsed_features)-1] + ' AS "' + raw_paths[len(raw_paths)-1] + '"'
```

Comment:
@alastair I had trouble passing the list of features into the sql query without concatenating them all into a string like this, and obviously the above is hard to follow. Is it preferable to use str.join(), or a different method to concatenate?